docs/proposals/29-autoscaling-storage-volumes.md

```diff
@@ -0,0 +1,335 @@
+
+_Fig.1: Runtime structure_
+
+For the role of a storage scaling controller, the existing, recently implemented Gardener PVC autoscaler application
+[[1]], hereafter referred to as `pvc-autoscaler`, is used. It is
+integrated as a seed system component, and runs as a deployment in the seed's `garden` namespace.
+Its primary driving signal is the PVC metrics from the seed's cache Prometheus instance, which it
+periodically examines, and if capacity is found to be near exhaustion, `pvc-autoscaler` takes action by updating the
+PVC's storage request.
+
+In this initial iteration, `pvc-autoscaler` scales two categories of `vali` and `prometheus` volumes:
+those in shoot namespaces, and those in the seed's `garden` namespace.
+
+`pvc-autoscaler` publishes Prometheus metrics to anonymous scrapers via a `/metrics` HTTP endpoint. A `ServiceMonitor`
+object is created in the `garden` namespace, and drives the seed's `prometheus-operator` to configure a scrape on that
+endpoint by `prometheus-seed`.
+
+`pvc-autoscaler` runs in active/passive replication mode, based on the standard leader election mechanism, supplied by
+the K8s controller runtime library. Only the active replica is shown in _Fig.1_.
+
+### PVC Autoscaler - Deployer Structure
+The `pvc-autoscaler` application and its supporting artefacts are deployed by `gardener`, as part of the seed
+reconciliation flow.
+
+The `vali` and `prometheus` deployers, part of `gardenlet`'s shoot reconciliation flow, ensure that the PVC template section in
+the respective StatefulSet objects contains the annotations, necessary to instruct `pvc-autoscaler` how to act on the
+PVCs, created from those templates.
+
+That `gardenlet` action mechanism, based on PVC templates, only affects PVCs upon creation. Therefore, it has no effect
+on existing shoots, and on the observability PVCs of existing seed clusters. To enable PVC autoscaling for existing
+shoots, those PVCs will be directly annotated by the operator. Such direct intervention allows gradual transition,
+where autoscaling can initially be enabled only for a few PVCs for validation purposes.
+Similarly, to enable PVC autoscaling on existing seed clusters' `garden` PVCs, those will be directly annotated at
+operators' discretion.
+For details regarding the transition of existing volumes to autoscaling, see 
+[section 'Enabling Autoscaling for Existing Volumes'](#enabling-autoscaling-for-existing-volumes)
+
+![02-operator_structure.png](assets/gep29-02-operator_structure.png)
+
+_Fig.2: Operator structure_
+
+VPA is used to scale `pvc-autoscaler`. For the future security enhancement scenario, where `kube-rbac-proxy` is used
+(see [section 'Future Enhancements'](#future-enhancements)),
+that container is expected to be excluded from autoscaling, and to run with fixed resource requests. Pod evictions
+driven by that lean container are unjustified, as they would only achieve minimal resource savings, and disrupt
+the main container.
```

Comment:
Too much detail indeed. The reason this is here, is because this GEP started as a different type of document: a design specification.

All I can do under the circumstances is, delete it. I'd hate to lose design details, but we currently don't have a good place for them, if not in the GEP.

Reply:
I extracted the secure metrics section into external document. I removed the details from here.